data/publisher/src/main/java/tech/pegasys/teku/data/publisher/MetricsPublisher.java

```diff
@@ -26,22 +28,23 @@
 public class MetricsPublisher {
 
   private static final Logger LOG = LogManager.getLogger();
+  private static final MediaType MEDIA_TYPE = MediaType.parse("application/json; charset=utf-8");
   private final OkHttpClient client;
-  private final MediaType mediaType = MediaType.parse("application/json; charset=utf-8");
+  private final HttpUrl endpoint;
 
-  public MetricsPublisher(final OkHttpClient client) {
+  public MetricsPublisher(final OkHttpClient client, final HttpUrl endpoint) {
     this.client = client;
+    this.endpoint = endpoint;
   }
 
-  public int publishMetrics(final String endpointAddress, final String json) throws IOException {
-    HttpUrl endpoint = HttpUrl.get(endpointAddress);
-    RequestBody body = RequestBody.create(json, mediaType);
+  public int publishMetrics(final String json) throws IOException {
+    RequestBody body = RequestBody.create(json, MEDIA_TYPE);
     Request request = new Request.Builder().url(endpoint).post(body).build();
     Response response = this.client.newCall(request).execute();
     int responseCode = response.code();
     response.close();
-    if (responseCode != 200) {
-      LOG.error(
+    if (responseCode != SC_OK) {
+      LOG.debug(
```

Comment:
I think we probably want this above debug level since it means we're failing to publish metrics you're likely to be interested in.  Probably should be either WARN or ERROR and it is indicative of an actual failure so probably ERROR is right.

Reply:
just that its overkill having several levels of errors i was thinking... will take another look

Reply:
If it is being caught and handled elsewhere then we should just remove this one, but my impression was this was the only place that checked the return type.

The other option would be to throw an exception when the response code is an error but I think that will be noisier than we really want.